src/auth/auth.service.ts

```diff
@@ -63,19 +65,21 @@ export class AuthService {
 		return this.usersService.createUser(payload);
 	}
 
-	async signIn(req: SignInRequestDto): Promise<string> {
+	async signIn(req: SignInRequestDto, type: RoleEnum): Promise<string> {
 		const user = await this.usersRepository.findBy({ email: req.email });
 		if (!user) {
 			throw new NotFoundException('Email or password is not correct');
 		} else if (!user.isActive) {
 			throw new BadRequestException('User is not active');
 		}
-
 		const checkPassword = await compare(req.password, user.password);
 		if (!checkPassword) {
 			throw new BadRequestException('Email or password is not correct');
 		}
 
+		const role = await this.roleRepository.findByName(type);
+		await this.roleRepository.addUserRole(user.id, role.id);
+
```

Comment:
Role must be assigned to the user on signUp route or super admin can assigned role to user.